tools/macos_buildenv.sh

```diff
@@ -93,7 +93,18 @@ case "$1" in
         export X_VCPKG_APPLOCAL_DEPS_INSTALL=ON
         export CMAKE_GENERATOR=Ninja
 
+        # We have reports that cmake fails to find a suitable compiler on the path
+        # of some installations. As a workaround we set always CC and CCX.
+        if which clang++-11 >/dev/null 2>&1; then
+          clang++ --version
+          echo "clang++ found TODO: Setting CC CXX should be not required. Remove?"
+        fi
```

Comment:
Please remove this. This script is designed to be run in ~/.zprofile or ~/.profile.

Reply:
Really? I am in doubt that any Mac user uses it like this. The are other echo messages as well.

Anyway, this is only a temporary solution until we have found withoutthe root cause that requires hard coding the compiler path. Without it we will never find out why this is necessary.

Please post the output the next time you are using you Mac. Also the error message would be nice that appears without setting CC and CXX.

Reply:
Hardcoding the compiler using CXX might be really future issue that is known to confuse xcode. It uses C++ instead of clang++. Now the output is: 
```
2021-09-09T14:54:28.3956930Z Apple clang version 12.0.0 (clang-1200.0.32.29)
2021-09-09T14:54:28.3960560Z Target: x86_64-apple-darwin19.6.0
2021-09-09T14:54:28.3961550Z Thread model: posix
2021-09-09T14:54:28.3963520Z InstalledDir: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
2021-09-09T14:54:28.3965780Z clang++ found TODO: Setting CC CXX should be not required. Remove?
2021-09-09T14:54:28.4048420Z ##[group]Run ccache  -M 2097152
2021-09-09T14:54:28.4048990Z [36;1mccache  -M 2097152[0m
2021-09-09T14:54:28.4219450Z shell: /bin/bash -e {0}
2021-09-09T14:54:28.4219880Z env:
2021-09-09T14:54:28.4220530Z   SCCACHE_COMMIT: 3f318a8675e4c3de4f5e8ab2d086189f2ae5f5cf
2021-09-09T14:54:28.4221420Z   APPLE_CODESIGN_IDENTITY: 2C2B5D3EDCE82BA55E22E9A67F16F8D03E390870
2021-09-09T14:54:28.4222210Z   MACOS_CODESIGN_OPENSSL_PASSWORD: 
2021-09-09T14:54:28.4222810Z   MACOS_CODESIGN_CERTIFICATE_PASSWORD: 
2021-09-09T14:54:28.4223480Z   BUILDENV_NAME: mixxx-deps-2.4-x64-osx-d36133a
2021-09-09T14:54:28.4224510Z   CC: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
2021-09-09T14:54:28.4225730Z   CXX: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ 
```
snip
```
2021-09-09T14:55:05.4779050Z -- The C compiler identification is AppleClang 12.0.0.12000032
2021-09-09T14:55:05.6149740Z -- The CXX compiler identification is AppleClang 12.0.0.12000032
2021-09-09T14:55:05.6453420Z -- Check for working C compiler: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
2021-09-09T14:55:05.9583730Z -- Check for working C compiler: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
2021-09-09T14:55:05.6453420Z -- Check for working C compiler: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
2021-09-09T14:55:05.9583730Z -- Check for working C compiler: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
2021-09-09T14:55:05.9684970Z -- Detecting C compiler ABI info
2021-09-09T14:55:06.1905530Z -- Detecting C compiler ABI info - done
2021-09-09T14:55:06.2168160Z -- Detecting C compile features
2021-09-09T14:55:06.2250120Z -- Detecting C compile features - done
2021-09-09T14:55:06.2353250Z -- Check for working CXX compiler: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
2021-09-09T14:55:06.4505310Z -- Check for working CXX compiler: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
```

Reply:
Please just put it back exactly how it was. I am tired of discussing this.

Reply:
Please have a close look to the code now. 

The requested code is there exactly as it was! 

In addition we have a debug message that should reveal the root cause of the issue once you find time to go back to your mac without any extra effort. In addition we have the chance one other finds the message and helps out here, if you don't find time. What's the draw back? What are alternatives?

Reply:
It is not exactly how it was. You have added an annoying print statement will print every time the user starts a new shell.

Reply:
Ping. Please remove this annoying print statement.